### PR TITLE
[Eigen3] Eigen update 

### PR DESCRIPTION
**Pull request by eclipse0922**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
~~- [ ] The "supports" clause reflects platforms that may be fixed by this new version~~
~~- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

fix-vectorized-reductions-half.patch is not needed anymore  https://gitlab.com/libeigen/eigen/-/commit/d0e3791b1a0e2db9edd5f1d1befdb2ac5a40efe0

### CODE REVIEW

File: ports/eigen3/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "eigen3",
   "version": "3.4.0",
```
Comment (Neumann-A) [inline ports/eigen3/vcpkg.json:3]: this seems incorrect if you change the sha.

**Comment by eclipse0922** [comment]
😭  As expected, nothing is easy.

File: ports/eigen3/vcpkg.json (hunk 1 above)
Comment (eclipse0922) [reply ports/eigen3/vcpkg.json:3]: There is no official release after 3.4.0.
just minor fixes.
Should I use version-date?

Comment (Neumann-A) [reply ports/eigen3/vcpkg.json:3]: version-date or keep the eigen update like me in your own overlay

**Comment by eclipse0922** [comment]
For someone who has interest, This upstream commit breaks many ports,
https://gitlab.com/libeigen/eigen/-/commit/c20e908ebc42f7174b1d85ce82583a66d185520c

File: ports/eigen3/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "eigen3",
-  "version": "3.4.0",
-  "port-version": 2,
+  "version-date": "2021-11-15",
```
Comment (Neumann-A) [inline ports/eigen3/vcpkg.json:3]: What why? It is 2023

Comment (eclipse0922) [reply ports/eigen3/vcpkg.json:3]: I tried latest commit, but the upstream commit(https://gitlab.com/libeigen/eigen/-/commit/c20e908ebc42f7174b1d85ce82583a66d185520c) breaks many things.

I hope I can fix all of them, But realistically, it's hard to fix everything.
The RBDL port is easy to fix, but the rest of the library is not,

Comment (eclipse0922) [reply ports/eigen3/vcpkg.json:3]: Originally I wanted to remove some warnings caused by old compiler options,

File: ports/rbdl/remove_check_template_params.patch
Comment (eclipse0922) [inline ports/rbdl/remove_check_template_params.patch:1]: `_check_template_params()` is removed.

https://gitlab.com/libeigen/eigen/-/merge_requests/647

**Comment by eclipse0922** [comment]
I will close this PR
In order to fix the build failure of other libraries caused by Eigen changes, I need to create a redundant patch. This patch has already been merged into Eigen upstream. This looks really awkard.
I will attempt to rectify any issues with other libraries and eigen upstream before proceeding to initiate a pull request for vcpkg ports.

**Pull request closed without merging**